src/test/java/teammates/logic/core/AccountsLogicTest.java

```diff
@@ -194,6 +196,17 @@ public void testJoinCourseForStudent() throws Exception {
                 () -> accountsLogic.joinCourseForStudent(finalStudent.getKey(), "wrongstudent"));
         assertEquals("Student has already joined course", eaee.getMessage());
 
+        ______TS("failure: course deleted");
+
+        studentData.setCourse(deletedCourseId);
+        studentData = studentsLogic.createStudent(studentData);
+        StudentAttributes finalDeletedCourseStudent = studentData;
```

Comment:
Any particular reason for reassigning? Seems a little out of place to me considering it's not reassigned for the other test cases. Also might be confusing since both variables now refer to the same object.

Can also consider adding a case for this in `JoinCourseActionTest.java`

Reply:
the reassignment was done previously at line 135
```
studentsLogic.createStudent(studentData);
        studentData = studentsLogic.getStudentForEmail(courseId,
                originalEmail);
        StudentAttributes finalStudent = studentData;
```
so I followed that example, should I do it this way?

Reply:
I personally think it's confusing, but no strong objections from me.